### ISSUE TITLE
[Accessibility] Inability to Interact with Map View Filters for Keyboard-Only Users

### ISSUE THREAD

**Issue by kivuvarosekivuvan** [ushahidi/platform#4968]
## Describe the bug

* The current filters in the "Map View" (Status, Categories, Date range, Location) are not keyboard-focusable and operable. Users are unable to choose a range for the calendar filter or tick the checkboxes using the keyboard alone.
* This being the parent Issue we have sub-issues inside it:
               [https://github.com/ushahidi/platform/issues/4967](https://github.com/ushahidi/platform/issues/4967)
               [https://github.com/ushahidi/platform/issues/4966](https://github.com/ushahidi/platform/issues/4966)
               [https://github.com/ushahidi/platform/issues/4965](https://github.com/ushahidi/platform/issues/4965)
               [https://github.com/ushahidi/platform/issues/4964](https://github.com/ushahidi/platform/issues/4964)
               [https://github.com/ushahidi/platform/issues/4963](https://github.com/ushahidi/platform/issues/4963)

## To Reproduce

1. Launch the local host
2. Navigate to the "Map View"
3. Find the filters under the Map View and try to interact with them individually without using the mouse.
4. Observe if you are able to complete these actions using the keyboard only.

## Expected behavior

Each filter should be keyboard-focusable and operable, allowing users to interact with them, including setting a date range in the calendar filter and ticking the checkboxes, without the use of a mouse

## Screenshots

![Screenshot from 2024-06-18 09-41-01](https://uploads.linear.app/f1df6e34-3bca-4ade-bb48-e6a457e64091/2c229df1-2a8d-4f21-820f-0b4ad3d72c84/7ac7fdeb-e9e1-46b0-8de0-1a7402a1f387?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2YxZGY2ZTM0LTNiY2EtNGFkZS1iYjQ4LWU2YTQ1N2U2NDA5MS8yYzIyOWRmMS0yYThkLTRmMjEtODIwZi0wYjRhZDNkNzJjODQvN2FjN2ZkZWItZTllMS00NmIwLThkZTAtMWE3NDAyYTFmMzg3IiwiaWF0IjoxNzE4OTYzMzkyLCJleHAiOjMzMjg5NTIzMzkyfQ.8afUuf8noTwCv_Kfd6cDbumJ-QP79Gg33JsVt59eKB0)

**Comment by Angamanga** [comment]
Closing but keeping in Linear: https://linear.app/ushahidi/issue/USH-1434/[accessibility]-inability-to-interact-with-map-view-filters-for
